live-ingester/src/main/java/org/alfresco/hxi_connector/live_ingester/adapters/messaging/transform/response/ATSTransformResponseHandler.java

```diff
@@ -51,33 +55,66 @@ public class ATSTransformResponseHandler extends RouteBuilder
 
     @Override
     public void configure()
     {
-        // @formatter:off
+        onException(EmptyRenditionException.class, ResourceNotFoundException.class)
+                .process(this::retryContentIngestion);
+
+        onException(Exception.class)
+                .log(ERROR, log, "Retrying ${routeId}, attempt ${header.CamelRedeliveryCounter} due to ${exception.message}")
+                .maximumRedeliveries(integrationProperties.alfresco().transform().response().retryIngestion().attempts())
+                .redeliveryDelay(integrationProperties.alfresco().transform().response().retryIngestion().initialDelay())
+                .backOffMultiplier(integrationProperties.alfresco().transform().response().retryIngestion().delayMultiplier());
+
         SecurityContext securityContext = SecurityContextHolder.getContext();
         from(integrationProperties.alfresco().transform().response().endpoint())
                 .routeId(ROUTE_ID)
-                .log(DEBUG, "Received transform completed event : ${body}")
+                .log(DEBUG, log, "Received transform completed event : ${body}")
                 .unmarshal()
                 .json(JsonLibrary.Jackson, TransformResponse.class)
                 .process(exchange -> SecurityContextHolder.setContext(securityContext))
-                .doTry()
-                    .process(this::ingestContent)
-                .doCatch(TransformationFailedException.class)
-                    .log(DEBUG, log, "Transformation failed: ${exception.message}")
+                .process(this::ingestContent)
                 .end();
-        // @formatter:on
     }
 
     @SuppressWarnings("PMD.UnusedPrivateMethod")
     private void ingestContent(Exchange exchange)
     {
         TransformResponse transformResponse = exchange.getIn().getBody(TransformResponse.class);
-        IngestContentCommand command = new IngestContentCommand(transformResponse.status(), transformResponse.targetReference(), transformResponse.clientData().nodeRef());
+
+        if (transformResponse.status() == 400)
+        {
+            log.atDebug().log("Rendition of node {} failed with status {}. Details: {}", transformResponse.clientData().nodeRef(), transformResponse.status(), transformResponse.errorDetails());
+            return;
+        }
+
+        IngestContentCommand command = new IngestContentCommand(transformResponse.targetReference(), transformResponse.clientData().nodeRef());
 
         ingestContentCommandHandler.handle(command);
     }
 
+    @SuppressWarnings("PMD.UnusedPrivateMethod")
+    private void retryContentIngestion(Exchange exchange)
+    {
+        Exception exception = exchange.getProperty(Exchange.EXCEPTION_CAUGHT, Exception.class);
+        TransformResponse transformResponse = exchange.getIn().getBody(TransformResponse.class);
+
+        int maxAttempts = integrationProperties.alfresco().transform().response().retryTransformation().attempts();
```

Comment:
This is transformation or ingestion retry? Method name indicates that ingestion but body is about transformation 🤔

Reply:
good point - it is transformation retry, I changed method name